### ISSUE TITLE
Searching

### ISSUE THREAD

**Issue by irfanrosly** [sohobloo/react-native-modal-dropdown#64]
How would if search is implemented? For example, user can type in the "Please select.." and it can filter the option. It would be nice for an option with a long list.

**Comment by sohobloo** [comment]
Good idea. I'll think about it.

**Comment by dmr07** [comment]
I agree. I'm trying to implement search with this module and there are a few things that make it difficult. For example, I call `modal_dropdown.show()` after every debounced key event, and I when I do the modal seems to steals focus the input area, causing the keyboard to close. The end result is the user cannot type more than one key at a time.

Setting keyboardShouldPersistTaps to true does not seem to help in this case.

**Comment by bradmatt275** [comment]
I'm quite new to React so this is far from perfect, but i was able to get it working with this code.

[CustomPicker.txt](https://github.com/sohobloo/react-native-modal-dropdown/files/1172706/CustomPicker.txt)

![screenshot01](https://user-images.githubusercontent.com/16756004/28566021-89cbca20-7160-11e7-9f7e-b08de5d7eec8.png)
![screenshot02](https://user-images.githubusercontent.com/16756004/28566025-8bda2d84-7160-11e7-8f9b-a5e6e7990345.png)

**Comment by sohobloo** [comment]
@bradmatt275 
We have to design searching function from both flexibility and friendly interactivity.
For flexibility, I think we can provide both a default searching function like your code and a custom interface for array options that we can not handle or some custom searching requirement.
For UI, I haven't found a nice way so for but yours give me inspiration.

Thanks.

**Comment by naomipol** [comment]
@dmr07 I'm also facing the same issue where keyboard dismissed when calling the `show()` method.
Have you reached any solution yet?
Thanks in advanced.

**Comment by cheesest3ak** [comment]
@naomipol have you found a solution for this?

**Comment by naomipol** [comment]
@cheesest3ak Iv'e ended up using a different library for this use case: https://github.com/l-urence/react-native-autocomplete-input

**Comment by mchen37** [comment]
I got something working based on @bradmatt275's great example. I filter the options data set myself based on the search.

Would it be possible to lock the first row, like have it sticky at the top? Currently if you scroll the first row out of view, the user will no longer be able to search.
